lib/util/http/host-rules.ts

```diff
@@ -107,6 +108,18 @@ export function applyHostRules(url: string, inOptions: GotOptions): GotOptions {
   if (!hasProxy() && foundRules.enableHttp2 === true) {
     options.http2 = true;
   }
+
+  if (foundRules.keyStore) {
+    let ksData = fs.readFileSync(foundRules.keyStore.filename);
```

Comment:
No

Reply:
Reading small certificate archives synchronously doesn't seem too bad. We're talking <4K files, they'd be read in two syscalls. I can refactor the call chain to be async if that's more desirable, and drop in a cache.

Reply:
Never use sync. And reading it from disk for every single request is highly undesirable too. Let's discuss in the issue before returning to this PR.